Python/Auto_Caption_Generator/caption_generator_Inception.py

```diff
@@ -0,0 +1,555 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# IMPORTS
+
+import numpy as np
+from numpy import array
+import pandas as pd
+import matplotlib.pyplot as plt
+import string
+import os
+from PIL import Image
+import glob
+from pickle import dump, load
+from time import time
+from keras.preprocessing import sequence
+from keras.models import Sequential
+from keras.layers import LSTM, Embedding, TimeDistributed, Dense, RepeatVector, Activation, \
+    Flatten, Reshape, concatenate, Dropout, BatchNormalization
+from keras.optimizers import Adam, RMSprop
+from keras.layers.wrappers import Bidirectional
+from keras.layers.merge import add
+from keras.applications.inception_v3 import InceptionV3
+from keras.preprocessing import image
+from keras.models import Model
+from keras import Input, layers
+from keras import optimizers
+from keras.applications.inception_v3 import preprocess_input
+from keras.preprocessing.text import Tokenizer
+from keras.preprocessing.sequence import pad_sequences
+from keras.utils import to_categorical
+
+# GPU Physical Growth
+import tensorflow as tf
+from tqdm.notebook import tqdm
+
+physical_devices = tf.config.list_physical_devices('GPU')
+tf.config.experimental.set_memory_growth(physical_devices[0], True)
+
+
+# load doc into memory
+def load_doc(filename):
+    """
+    Function for loading various files
+    :param filename: Source Path
+    :return: Caption Text
+    """
+    # open the file as read only
+    file = open(filename, 'r')
+    # read all text
+    text = file.read()
+    # close the file
+    file.close()
+    return text
+
+
+# Load Caption Datatset
+filename = "Flickr8k_text/Flickr8k.token.txt"
+doc = load_doc(filename)
+
+
+def load_descriptions(doc):
+    """
+    Loads description (Captions)
+    :param doc: Caption text files
+    :return: Captions mapped to images
+    """
+    mapping = dict()
+    # process lines
+    for line in doc.split('\n'):
+
+        # split line by white space
+        tokens = line.split()
+        if len(line) < 2:
+            continue
+
+        # take the first token as the image id, the rest as the description
+        image_id, image_desc = tokens[0], tokens[1:]
+
+        # extract filename from image id
+        image_id = image_id.split('.')[0]
+
+        # convert description tokens back to string
+        image_desc = ' '.join(image_desc)
+
+        # create the list if needed
+        if image_id not in mapping:
+            mapping[image_id] = list()
+
+        # store description
+        mapping[image_id].append(image_desc)
+    return mapping
+
+
+# parse descriptions
+descriptions = load_descriptions(doc)
+print('Loaded: %d ' % len(descriptions))
+
+
+def clean_descriptions(descriptions):
+    """
+    FUnction to Remove Extra  whitespaces, punctuation marks etc
```

Comment:
Replace `FUnction` to `Function`

Reply:
While I was at this, I removed a few more spelling mistakes as well.